internal/pkg/aws/ecs/ecs.go

```diff
@@ -226,19 +228,29 @@ func (e *ECS) RunTask(input RunTaskInput) ([]*Task, error) {
 		taskARNs[idx] = aws.StringValue(task.TaskArn)
 	}
 
-	if err := e.client.WaitUntilTasksRunning(&ecs.DescribeTasksInput{
+	err = e.client.WaitUntilTasksRunning(&ecs.DescribeTasksInput{
 		Cluster: aws.String(input.Cluster),
 		Tasks:   aws.StringSlice(taskARNs),
-	}); err != nil {
-		return nil, fmt.Errorf("wait for tasks to be running: %w", err)
+	})
+	if err == nil {
```

Comment:
what do you think of this flow:
```go
runErr := e.client.WaitUntilTasksRunning(...)
if !isRequestTimeoutErr(runErr) {
      // Unexpected error, tasks could not run.
      return nil, fmt.Errorf("wait for tasks to be running: %w", runErr)
}

tasks, describeErr := e.DescribeTasks(input.Cluster, taskARNs)
if describeErr != nil {
       return nil, describeErr
}
if runErr != nil { // tasks timed out, return a description of stopped tasks.
     return nil, &ErrWaiterResourceNotReadyForTasks{tasks: tasks, awsErrResourceNotReady: runErr}
}
return tasks, nil
```
where the `aerr` stuff is moved to its own function.